lib/vmdb/settings.rb

```diff
@@ -198,5 +206,34 @@ def self.apply_settings_changes(resource, deltas)
       end
     end
     private_class_method :apply_settings_changes
+
+    def self.process_magic_value(resource, delta)
+      return false unless MAGIC_VALUES.include?(delta[:value])
+      case delta[:value]
+      when DELETE_COMMAND
+        _log.info("removing custom settings #{delta[:key]} on #{resource.class} id:#{resource.id}")
+        resource.settings_changes.where("key LIKE ?", "#{delta[:key]}%").destroy_all
+      when DELETE_ALL_COMMAND
+        _log.info("resetting #{delta[:key]} to defaul for all resorces")
+        SettingsChange.where("key LIKE ?", "#{delta[:key]}%").destroy_all
+      end
+      true
+    end
+    private_class_method :process_magic_value
+
+    def self.walk_hash(hash, path = [], &block)
+      hash.each do |key, value|
+        key_path = path.dup << key
+        yield key, value, key_path, hash
+        walk_hash(value, key_path, &block) if value&.class == hash.class
```

Comment:
Won't `hash.class` just always be `Hash`? Also do we want to recurse into something like an array of hashes? I think this would fail in that case.

Reply:
@carbonin it is always ```Hash```,  will change.
Array of Hashes would be OK, I am adding another rspec test for it ...